src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt

```diff
@@ -218,6 +221,36 @@ class EnterpriseEnhancedContextPanel(project: Project, chatSession: ChatSession)
     return toolbar.createPanel()
   }
 
+  override fun createCheckboxPolicy(): CheckboxTreeBase.CheckPolicy =
+      CheckboxTreeBase.CheckPolicy(
+          /* checkChildrenWithCheckedParent = */ false,
+          /* uncheckChildrenWithUncheckedParent = */ false,
+          /* checkParentWithCheckedChild = */ false,
```

Comment:
since these changed we will need to double check the check-uncheck behavior

Reply:
This behavior has only changed for the enterprise context panel, and this is the behavior we need.

Without this change, turning off enhanced context unchecks the repos, *which removes the repos.* That's not how the extension works:

- There's a set of repos: zero or one implicitly included repos, plus explicitly chosen repos up to a limit of 10.
- There's enhanced context toggle, which controls fetching from the set of repos and the open tabs in the editor.
- You can control the repo setting and the enhanced context toggle independently.

Setting these all to false turns off the automatic checking/unchecking.